notification-bot/src/index.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  Filter,
+  StarkNetCursor,
+  v1alpha2,
+  FieldElement,
+  EventFilter,
+  FilterBuilder,
+} from "@apibara/starknet";
+
+import { RpcProvider, constants, number, provider, uint256 } from "starknet";
+import { formatUnits } from "ethers";
+import * as dotenv from "dotenv";
+import { BlockNumber } from "starknet";
+import * as fs from "fs";
+import * as toml from "toml";
+import { Logger } from 'logger';
+dotenv.config();
+
+const configPath = process.env.CONFIG_PATH || 'config.toml';
+const config = toml.parse(fs.readFileSync(configPath, 'utf-8'));
+
+const BOT_API_KEY = config.telegram?.bot_api_key;
+const CHATID = config.telegram?.chat_id;
+const CLIENT_URL = config.apibara?.url;
+const CLIENT_TOKEN = config.apibara?.token;
+const GOV_CONTRACT_ADDRESS = config.governance?.gov_contract_address;
+const PROPOSAL_HASH = "0x01b5f21c50bf3288fb310446824298a349f0ed9e28fb480cc9a4d54d034652e1"
+
+
+/**
+ * Initializes the StreamClient with configuration loaded from environment variables.
+ * @returns {StreamClient} Configured StreamClient instance for connecting to Apibara.
+ */
+function initializeStreamClient() {
+  return new StreamClient({
+    url: CLIENT_URL,
+    token: CLIENT_TOKEN,
+    async onReconnect(err, retryCount) {
+      console.log("reconnect", err, retryCount);
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      return { reconnect: true };
+    },
+  });
+}
+
+/**
+ * Initializes the event filter configuration for the Apibara client.
+ * @returns {string} Encoded filter string used to specify which blockchain events to listen to.
+ */
+function initializeFilter() {
+  const governance_contract_address = FieldElement.fromBigInt(BigInt(GOV_CONTRACT_ADDRESS));
+  return Filter.create()
+    .withHeader({ weak: true })
+    .addEvent(new EventFilter().withFromAddress(governance_contract_address))
+    .encode();
+}
+
+/**
+ * Configures the Apibara client with a specific filter and block number.
+ * @param {StreamClient} client The Apibara client instance to configure.
+ * @param {string} filter The encoded filter specifying the events to listen for.
+ * @param {number} block_number The blockchain block number from which to start listening.
+ */
+function configureClient(client: StreamClient, filter: Uint8Array, block_number: number) {
+  client.configure({
+    filter: filter,
+    batchSize: 1,
+    cursor: StarkNetCursor.createWithBlockNumber(block_number),
+  });
+}
+
+/**
+ * Main entry point for the event monitoring script.
+ * Initializes the StreamClient and RpcProvider for connecting to the StarkNet blockchain.
+ * Configures the Apibara client to listen for specific blockchain events and processes them accordingly.
+ */
+async function main() {
+  try {
+    // Apibara streaming
+    const client = initializeStreamClient();
+
+    const provider = new RpcProvider({
+      nodeUrl: constants.NetworkName.SN_SEPOLIA,
+      chainId: constants.StarknetChainId.SN_SEPOLIA,
```

Comment:
And chainId too, in config.toml. The message could contain information about from which network the proposal was proposed from.